src/Compilers/Shared/NamedPipeUtil.cs

```diff
@@ -16,15 +16,36 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     internal static class NamedPipeUtil
     {
+        // Size of the buffers to use: 64K
+        private const int PipeBufferSize = 0x10000;
+
+        private static string GetPipeNameOrPath(string pipeName)
+        {
+            if (PlatformInformation.IsUnix)
+            {
+                // If we're on a Unix machine then named pipes are implemented using Unix Domain Sockets.
+                // Most Unix systems have a maximum path length limit for Unix Domain Sockets, with
+                // Mac having a particularly short one. Mac also has a generated temp directory that
+                // can be quite long, leaving very little room for the actual pipe name. Fortunately,
+                // '/tmp' is mandated by POSIX to always be a valid temp directory, so we can use that
+                // instead.
+                return Path.Combine("/tmp", pipeName);
```

Comment:
Didn't @kg mention that some mono scenarios didn't have a `/tmp` directory?

Reply:
I'm not aware of these situations, but someone else brought it up on one of my PRs

Reply:
Well, there are technically two things to separate here: whether /tmp exists, and whether it is the system temporary directory.

POSIX requires /tmp exist, so I think almost all reasonable platforms should have it. 

However, no system is required to choose /tmp as their system temporary directory. The most prominent example here is macOS, where they have a randomly generated one.

In this case I'm assuming that even if /tmp isn't the *preferred* system temporary directory, it's a valid directory. I think this is a reasonable assumption and probably shouldn't have severe consequences otherwise. If not for the UDS path length limit I would just use the standard system temporary directory.